src/NzbDrone.Core.Test/DecisionEngineTests/UpgradeAllowedSpecificationFixture.cs

```diff
@@ -0,0 +1,293 @@
+                Language.English
+            ).Should().BeFalse();
+        }
+
+        [Test]
+        public void should_return_true_for_language_upgrade_when_upgrading_is_allowed()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = true
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English, Language.French),
+                    Cutoff = Language.French,
+                    UpgradeAllowed = true
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.English },
+                new QualityModel(Quality.MP3_320),
+                Language.French
+            ).Should().BeTrue();
+        }
+
+        [Test]
+        public void should_return_true_for_same_language_when_upgrading_is_allowed()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = true
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English, Language.French),
+                    Cutoff = Language.French,
+                    UpgradeAllowed = true
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.English },
+                new QualityModel(Quality.MP3_320),
+                Language.English
+            ).Should().BeTrue();
+        }
+
+        [Test]
+        public void should_return_true_for_same_language_when_upgrading_is_not_allowed()
+        {
+            Subject.IsUpgradeAllowed(
+                new QualityProfile
+                {
+                    Cutoff = Quality.FLAC.Id,
+                    Items = Qualities.QualityFixture.GetDefaultQualities(),
+                    UpgradeAllowed = true
+                },
+                new LanguageProfile
+                {
+                    Languages = LanguageFixture.GetDefaultLanguages(Language.English, Language.French),
+                    Cutoff = Language.French,
+                    UpgradeAllowed = false
+                },
+                new List<QualityModel> { new QualityModel(Quality.MP3_320) },
+                new List<Language> { Language.French },
+                new QualityModel(Quality.MP3_320),
+                Language.English
+            ).Should().BeTrue();
+        }
+
+        [Test]
+        public void should_return_true_for_lower_language_when_upgrading_is_allowed()
```

Comment:
Shouldn't a worse language return `false`?

Also, seems one of this or `should_return_true_for_language_upgrade_when_upgrading_is_allowed()` should return false.